src/qrules/combinatorics.py

```diff
@@ -21,13 +23,28 @@
     from qrules.particle import ParticleCollection
 
 
-StateWithSpins = tuple[str, Sequence[float]]
+StateWithSpins = tuple[str, Sequence[Fraction]]
 StateDefinition = Union[str, StateWithSpins]
 """Particle name, optionally with a list of spin projections."""
+StateDefinitionInput = Union[str, tuple[str, Sequence[Scalar]]]
+"""Input type for `StateDefinition` permitting also `int` and `float`"""
 InitialFacts = MutableTransition[ParticleWithSpin, InteractionProperties]
 """A `.Transition` with only initial and final state information."""
 
 
+def as_state_definition(
+    definition: StateDefinitionInput,
+) -> StateDefinition:
+    if type(definition) is str:
+        return definition
+    if type(definition) is tuple:
+        name = definition[0]
+        state = definition[1]
+        return name, list(map(Fraction, state))  # type: ignore  # noqa: PGH003
```

Comment:
Why was mypy complaining here?

Reply:
mypy doesn't recognize the constructor as a function and interpreted it as a type.